bindings/java/src/main/java/org/openucx/jucx/ucp/UcpMemory.java

```diff
@@ -71,10 +90,27 @@ public ByteBuffer getRemoteKeyBuffer() {
         return result;
     }
 
-    public ByteBuffer getData() {
+    public ByteBuffer getData() throws IOException {
+        if (data == null) {
+            try {
+                if (length > Integer.MAX_VALUE) {
+                    throw new UcxException("Can not get ByteBuffer of memory > 2GB");
+                }
+                data = (ByteBuffer)directBufferConstructor.newInstance(address, (int)length);
```

Comment:
it's weird that in some case we initialize the buffer in setData() and in other in setData()

Reply:
In 1 case user passed ByteBuffer object to register. In another case he may pass address and length, and may want to use ByteBuffer on top of this memory.